sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosContainer.java

```diff
@@ -347,7 +347,7 @@ private CosmosItemResponse mapDeleteItemResponseAndBlock(Mono<CosmosAsyncItemRes
      * @return the cosmos sync item response
      * @throws CosmosClientException the cosmos client exception
      */
-    public CosmosItemResponse deleteItem(String itemId, PartitionKey partitionKey,
+    public CosmosItemResponse<Object> deleteItem(String itemId, PartitionKey partitionKey,
```

Comment:
API change for @kushagraThapar and @moderakh to review.

Reply:
Same API as above.